### DISCUSSION TITLE
Git Prompts for UserCredentials Everytime

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#11333] [General] [answered]
Hi,
I'm currently deploying Theia in Eclipse Che. And am currently using these builtin vscode plugins to support Git. 
```
    "vscode-builtin-git": "https://open-vsx.org/api/vscode/git/1.52.1/file/vscode.git-1.52.1.vsix",
    "vscode-builtin-git-ui": "https://open-vsx.org/api/vscode/git-ui/1.55.2/file/vscode.git-ui-1.55.2.vsix"
```
However, every time I make a push, pull Git prompts for username and password. Is there any recommended way to store these credentials in the Che ? Am I missing something ?

Thanks in Advance

**Comment by vince-fugnitto** [comment, marked as answer]
@maheshrajrp thank you for the discussion, I believe it is related to the extensions themselves and your configuration and not Theia.

The following information may be useful, from what I understand the username and pass are not stored by the extension so you'll need to set up a credentials manager:
- https://code.visualstudio.com/Docs/editor/versioncontrol#_remotes
- https://stackoverflow.com/a/34627954
